app/models/host/discovered.rb

```diff
@@ -112,6 +113,18 @@ def refresh_facts
     return self.class.import_host_and_facts facts
   end
 
+  def reboot
+    logger.info "ForemanDiscovery: Rebooting #{name}"
+    status = ::ProxyAPI::BMC.new(:url => proxy_url).power :action => "cycle"
```

Comment:
@orrabin Only if the reboot is done via smart-proxy.
If it's directly to the host you need to use the code above (```::ProxyAPI::BMC.new(:url => proxy_url).power :action => "cycle"```). If it's through the smart proxy ```PUT <proxy_url>/discovery/:ip/reboot```